### PR TITLE
(dropbox) Changed to use the offline installer, other small improvements.

### PR DESCRIPTION
**Pull request by Boyan-Kostadinov**
Set Url to use the off-line installer
Changed package to be installed with "Install-ChocolateyPackage"
Removed dependency on AutoHotkey in favor of a compiled EXE that just closes the Dropbox window after install

### CODE REVIEW

File: automatic/dropbox/tools/chocolateyInstall.ps1 [hunk 1]
```diff
@@ -1,44 +1,11 @@
...
-  }
-
-} catch {
-
-  Write-ChocolateyFailure $packageName $($_.Exception.Message)
-  throw
-}
+Install $packageName $url $checksum
 
+Start-Process "$packagePath\Dropbox.exe"
```
Comment (ferventcoder) [inline automatic/dropbox/tools/chocolateyInstall.ps1]: What is this file?

Comment (ferventcoder) [inline automatic/dropbox/tools/chocolateyInstall.ps1]: Will need the source code for this. It's going to need to be reviewed. Users who are security minded will also want to see the source for this.

File: automatic/dropbox/dropbox.nuspec [hunk 2]
```diff
@@ -26,13 +26,5 @@ The confusing dialog box with the content *“Dropbox will not work correctly if
...
-        <dependency id="autohotkey.portable" />
-    </dependencies>
   </metadata>
-  <files>
-    <file src="tools\**" target="tools" />
-    <!--<file src="content\**" target="content" />-->
-  </files>
-</package>
-<!-- character encoding: “UTF-8” -->
+</package>
```
Comment (ferventcoder) [inline automatic/dropbox/dropbox.nuspec]: Looks like you may have accidentally removed only packing the files in the tools folder.

File: automatic/dropbox/dropbox.nuspec [hunk 3]
```diff
@@ -26,13 +26,5 @@ The confusing dialog box with the content *“Dropbox will not work correctly if
...
-    <dependencies>
-        <dependency id="autohotkey.portable" />
-    </dependencies>
   </metadata>
-  <files>
-    <file src="tools\**" target="tools" />
-    <!--<file src="content\**" target="content" />-->
-  </files>
-</package>
-<!-- character encoding: “UTF-8” -->
```
Comment (ferventcoder) [inline automatic/dropbox/dropbox.nuspec]: It looks like you may have accidentally removed only getting the tools folder packaged in.

File: automatic/dropbox/tools/Helpers.psm1 [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        $url `
+        $url `
+        -validExitCodes $validExitCodes `
+        -Checksum64 "$checksum" `
+        -ChecksumType64 $checksumType `
+        -ChecksumType "$checksumType" `
+        -Checksum "$checksum"
+}
+
+Export-ModuleMember *
```
Comment (ferventcoder) [inline automatic/dropbox/tools/Helpers.psm1]: What benefit (out of curiosity), does this abstraction provide?

File: automatic/dropbox/tools/Helpers.psm1 [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    $installerType = 'exe'
+    $validExitCodes = @(0)
+    $silentArgs = '/S'
+    $checksumType = 'sha256';
+
+    Install-ChocolateyPackage $packageName `
+        $installerType `
+        $silentArgs `
+        $url `
+        $url `
```
Comment (ferventcoder) [inline automatic/dropbox/tools/Helpers.psm1]: This little guy is possibly wrong here. Url should never be used for both unless it contains native installers for both 32 bit ad 64 bit.

File: automatic/dropbox/tools/chocolateyUninstall.ps1 [hunk 6]
```diff
@@ -1,19 +1,10 @@
...
-  }
-
-} catch {
-  Write-ChocolateyFailure $packageName $($_.Exception.Message)
-  throw
-}
+$packageName = 'dropbox'
+$fileType = 'exe'
+$silentArgs = '/S'
+$uninstallerPath = (GetDropboxRegProps).UninstallString
```
Comment (ferventcoder) [inline automatic/dropbox/tools/chocolateyUninstall.ps1]: This function looks like it was removed.

File: automatic/dropbox/tools/chocolateyInstall.ps1 (hunk 1 above)
Comment (Boyan-Kostadinov) [inline automatic/dropbox/tools/chocolateyInstall.ps1]: It's in there, it's the compiled version of the Dropbox.ahk file.

File: automatic/dropbox/dropbox.nuspec (hunk 3 above)
Comment (Boyan-Kostadinov) [inline automatic/dropbox/dropbox.nuspec]: Nope, that was on purpose, so it includes all the files in the package without them having to be inside the "tools" directory.

File: automatic/dropbox/tools/Helpers.psm1 (hunk 4 above)
Comment (Boyan-Kostadinov) [inline automatic/dropbox/tools/Helpers.psm1]: To you, probably none, I've been creating a bunch of packages, and having common functionality in one place means I can just re-use the helpers module across packages.

File: automatic/dropbox/dropbox.nuspec (hunk 3 above)
Comment (ferventcoder) [inline automatic/dropbox/dropbox.nuspec]: Fair, but typically if you are including more folders, you can specify those as well.

File: automatic/dropbox/tools/Helpers.psm1 (hunk 4 above)
Comment (ferventcoder) [inline automatic/dropbox/tools/Helpers.psm1]: You haven't read up on extensions yet have you? https://chocolatey.org/docs/how-to-create-extensions

Comment (ferventcoder) [inline automatic/dropbox/tools/Helpers.psm1]: The other is I would start with the most modern knowledge coming from the default template `choco new test` before attempting to improve on it. You may find that you don't even need this.

~~~powershell
$packageName= 'dropbox' 
$toolsDir   = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"

$packageArgs = @{
  packageName   = $packageName
  fileType      = 'exe'
  url           = 'https://clientupdates.dropboxstatic.com/client/Dropbox%2014.4.19%20Offline%20Installer.exe'

  softwareName  = 'Dropbox*'
  checksum      = 'A0827474448EF4EC4E65D61E00757F7812DF9E9004B6D944FDEDEDE6D88DEFE7'
  checksumType  = 'sha256'
  silentArgs    = '/S'           # NSIS
  validExitCodes= @(0)
}

Install-ChocolateyPackage @packageArgs
~~~

**Comment by Boyan-Kostadinov** [comment]
This is my first PR substitution. I appreciate your feedback, and I'm sorry for the bonehead mistakes. I should have spent more time reading the docs.

I didn't realize you were the main guy behind Chocolatey. You've done great work, I really admire what you've done with it.

**Comment by ferventcoder** [comment]
> This is my first PR submission so I appreciate your feedback, and I'm sorry for the bonehead mistakes.

@Boyan-Kostadinov no worries! We all have something to learn, I wouldn't even worry about it. 

I know there is another package you were looking at helping out with. We are currently moving most of my packages over to the coreteampackages repository so I can get help as there is so many things that pull me in all sorts of directions and fixes and general upkeep for my packages has fallen behind. Apologies :/

**Comment by Boyan-Kostadinov** [comment]
It's all good, when you get around to it. I have a working Spotify package that I wanted to contribute. Had to get sneaky to get that working.

**Comment by ferventcoder** [comment]
> It's all good, when you get around to it. I have a working Spotify package that I wanted to contribute. I had to get sneaky to get that working.

That's the one! I would have responded to the email but it's a bit of a conflict of interest so I'd rather leave it for another moderator to pick up.

**Comment by ferventcoder** [comment]
> I didn't realize you were the main guy behind Chocolatey. You've done great work, I admire what you've done with it.

Thanks! The community has really helped immensely in providing feedback, contributions, and community packages. We could not have done this alone. :D

**Comment by Boyan-Kostadinov** [comment]
This should be all set. Took all your suggestions. No need for the AHK script. Added choco extensions, creatively called "boyan-choco." Not sure if this makes sense. L:et me know when you get a chance to review.

**Comment by Boyan-Kostadinov** [comment]
I forgot to mention that through the extensions, the Dropbox package supports:
1. Install from a local setup/netowrk setup by passing the setup.exe
```
Dropbox -params '/setup="that\to\Dropbox 15.4.22 Offline Installer.exe"'
```
2. Or the package can be re-compiled including the offline installer inside it, and it will use that if not setup was specified.
3. Or it will use the Web installer if all else fails.

**Comment by Boyan-Kostadinov** [comment]
Looks like you updated your package, I'm going to close this PR.

**Pull request closed without merging**